### PR TITLE
Transport Layer streaming retries

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Add retries for client streaming queue and transport streaming queue.

### JIRA ticket:
https://rchain.atlassian.net/browse/CORE-1317

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

**Comment by sebtomba** [comment]
bors try

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 1]
```diff
@@ -46,7 +46,7 @@ class TcpTransportLayer(port: Int, cert: String, key: String, maxMessageSize: In
   private def certInputStream = new ByteArrayInputStream(cert.getBytes())
   private def keyInputStream  = new ByteArrayInputStream(key.getBytes())
 
-  private val streamObservable = new StreamObservable(100, tempFolder)
+  private val streamObservable = new StreamObservable(1000, tempFolder)
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: those values are arbitrary, please create ticket in jira to fine tune this at some point (maybe configuration flag would be a good opiton here)

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 2]
```diff
@@ -201,23 +201,37 @@ class TcpTransportLayer(port: Int, cert: String, key: String, maxMessageSize: In
...
+              .flatMap {
+                case Left(error) =>
+                  log.error(s"Error while streaming packet, error: $error") *> delay(
+                    handle(retryCount - 1)
+                  )
+                case Right(_) => Task.unit
+              }
+          case Left(error) => log.error(s"Error while streaming packet, error: $error")
+        } >>= kp(deleteFile(toStream.path).void)
+      else deleteFile(toStream.path).void
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: I would log on `debug` always that the stream was not handled
I would also log on `error` if delete a file was not a success

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 3]
```diff
@@ -187,6 +188,26 @@ class TcpTransportLayer(port: Int, cert: String, key: String, maxMessageSize: In
                })
     } yield result
 
+  private def deleteFile(path: Path): Task[Unit] = {
+    def delete(): Task[Unit] =
+      for {
+        result <- Task.delay(Try(path.toFile.delete).toEither)
```
Comment (rabbitonweb) [inline comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: Why not `Task.delay(path.toFil.delete).attempt` ?

Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: yes, absolutely!

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala (hunk 1 above)
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: I will prepare this value as a constructor parameter and create a ticket.

**Pull request merged**